Race condition about embed_genesis.cmake

### Build Error Description

When building with multiple cores, there is probably a race condition that multiple `embed_genesis.cmake` instances may be started at the same time and try to write to the same destination files, which leads to `configure_file Problem configuring file` errors.

### Build Environment
Github Autobuild.

### Console Logs (optional)
Ubuntu release build of https://github.com/bitshares/bitshares-core/runs/724112359
```
[ 65%] Linking CXX static library libgraphene_chain.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Built target graphene_chain
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
Scanning dependencies of target graphene_egenesis_none
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Building CXX object libraries/egenesis/CMakeFiles/graphene_egenesis_none.dir/egenesis_none.o
[ 65%] Linking CXX static library libgraphene_egenesis_none.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Built target graphene_egenesis_none
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
CMake Error at /home/runner/work/bitshares-core/bitshares-core/libraries/egenesis/embed_genesis.cmake:11 (configure_file):
  configure_file Problem configuring file


make[2]: *** [libraries/egenesis/egenesis_brief.cpp] Error 1
make[2]: *** Deleting file 'libraries/egenesis/egenesis_brief.cpp'
libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/build.make:80: recipe for target 'libraries/egenesis/egenesis_brief.cpp' failed
make[1]: *** [libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/all] Error 2
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[1]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:1266: recipe for target 'libraries/egenesis/CMakeFiles/graphene_egenesis_brief.dir/all' failed
Scanning dependencies of target graphene_egenesis_full
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make[2]: Entering directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 65%] Generating egenesis_brief.cpp, egenesis_full.cpp
-- Generating egenesis
-- Chain-id: 4018d7844c78f6a6c41c6a552b898022310fc5dec06da467ee7905a8dad512c8
[ 66%] Building CXX object libraries/egenesis/CMakeFiles/graphene_egenesis_full.dir/egenesis_full.o
[ 66%] Linking CXX static library libgraphene_egenesis_full.a
make[2]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
[ 66%] Built target graphene_egenesis_full
make[1]: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
make: *** [all] Error 2
Makefile:146: recipe for target 'all' failed
make: Leaving directory '/home/runner/work/bitshares-core/bitshares-core/_build'
##[error]Process completed with exit code 2.
```

## CORE TEAM TASK LIST
- [ ] Evaluate `Build Error`
- [ ] Provide build guidance
- [ ] <OR> Create `Bug Report`

--- a/libraries/egenesis/CMakeLists.txt
+++ b/libraries/egenesis/CMakeLists.txt
@@ -1,13 +1,11 @@
-message( STATUS "Generating egenesis" )
-
 if( GRAPHENE_EGENESIS_JSON )
   set( embed_genesis_args "${GRAPHENE_EGENESIS_JSON}" )
 else( GRAPHENE_EGENESIS_JSON )
   set( embed_genesis_args "genesis.json" )
 endif( GRAPHENE_EGENESIS_JSON )
 
-add_custom_command(
-   OUTPUT
+add_custom_target( build_egenesis_cpp
+   BYPRODUCTS
       "${CMAKE_CURRENT_BINARY_DIR}/egenesis_brief.cpp"
       "${CMAKE_CURRENT_BINARY_DIR}/egenesis_full.cpp"
    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
@@ -16,6 +14,7 @@ add_custom_command(
         -DINIT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
         -Dembed_genesis_args=${embed_genesis_args}
         -P ${CMAKE_CURRENT_SOURCE_DIR}/embed_genesis.cmake
+   COMMENT "Generating egenesis"
    DEPENDS
       "${GRAPHENE_EGENESIS_JSON}"
       "${CMAKE_CURRENT_SOURCE_DIR}/egenesis_brief.cpp.tmpl"
@@ -26,8 +25,10 @@ add_library( graphene_egenesis_none egenesis_none.cpp
              include/graphene/egenesis/egenesis.hpp )
 add_library( graphene_egenesis_brief "${CMAKE_CURRENT_BINARY_DIR}/egenesis_brief.cpp"
              include/graphene/egenesis/egenesis.hpp )
+add_dependencies( graphene_egenesis_brief build_egenesis_cpp )
 add_library( graphene_egenesis_full  "${CMAKE_CURRENT_BINARY_DIR}/egenesis_full.cpp"
              include/graphene/egenesis/egenesis.hpp )
+add_dependencies( graphene_egenesis_full build_egenesis_cpp )
 
 target_link_libraries( graphene_egenesis_none graphene_chain fc )
 target_link_libraries( graphene_egenesis_brief graphene_chain fc )

